src/main/java/org/opentripplanner/graph_builder/module/osm/OSMDatabase.java

```diff
@@ -958,6 +963,41 @@ private void processRoad(OSMRelation relation) {
         }
     }
 
+    /**
+     * Handle route=bicycle relations. Copies their network type to all way members.
+     *
+     * @param relation
+     * @author hbruch
+     * @see "https://wiki.openstreetmap.org/wiki/Tag:route%3Dbicycle"
+     */
+    private void processBicycleRoute(OSMRelation relation) {
+        String network = relation.getTag("network");
+        if (network == null)
+            network = "lcn";
+        switch (network) {
+            case "lcn": setNetworkForAllMembers(relation, "lcn"); break;
+            case "rcn": setNetworkForAllMembers(relation, "rcn"); break;
+            case "ncn": setNetworkForAllMembers(relation, "ncn"); break;
+            // we treat networks without known network type like local networks
+            default: setNetworkForAllMembers(relation, "lcn"); break;
+        }
+    }
+
+    private void setNetworkForAllMembers(OSMRelation relation, String lcn) {
+        for (OSMRelationMember member : relation.getMembers()) {
```

Comment:
Again, in my (functional programmer) view this would more easily if we used the streams API rather than `continue`. I admit that it's a style thing.